cmd/podman/start.go

```diff
@@ -108,7 +108,7 @@ func startCmd(c *cliconfig.StartValues) error {
 			}
 
 			// attach to the container and also start it not already running
-			err = startAttachCtr(ctr, os.Stdout, os.Stderr, inputStream, c.DetachKeys, sigProxy, !ctrRunning)
+			err = startAttachCtr(ctr, os.Stdout, os.Stderr, inputStream, c.DetachKeys, sigProxy, !ctrRunning, false)
```

Comment:
My thinking is podman pod start should be used in this case, so it should error out if the pod isn't started. The advantage of setting all of this up for run is the container is supposed to be created, then started in one command, so there isn't time to run pod start in between. adding this functionality to container start and having the infra container start makes less sense to me, and also requires a query to the state that otherwise isn't needed.

Reply:
Agree here

Reply:
@mheon concurrent commenting makes this confusing. Who are you agreeing with?

Reply:
@haircommander I think it's sensible to recursively start here if the container is in a pod

Reply:
@mheon @rhatdan fixed. Also made the change for podman attach, but opted to not change play_kube

Reply:
Agree with play kube, it should be doing ordered start already.

Reply:
(And, honestly, if it fails, it's because the dependency containers just don't exist yet, nevermind they aren't started)